src/main/java/org/terasology/climateConditions/ConditionsBaseField.java

```diff
@@ -15,41 +15,43 @@
  */
 package org.terasology.climateConditions;
 
-import com.google.common.base.Function;
 import org.terasology.math.TeraMath;
-import org.terasology.utilities.procedural.Noise2D;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.registry.In;
 import org.terasology.utilities.procedural.SimplexNoise;
+import org.terasology.world.WorldProvider;
 
 public class ConditionsBaseField {
-    private Noise2D noiseTable;
-    private int seaLevel;
-    private int maxLevel;
-    private float noiseMultiplier;
-    private Function<Float, Float> function;
-
-    public ConditionsBaseField(int seaLevel, int maxLevel, float noiseMultiplier,
-                               Function<Float, Float> function, long conditionSeed) {
-        this.seaLevel = seaLevel;
-        this.maxLevel = maxLevel;
-        this.noiseMultiplier = noiseMultiplier;
-        this.function = function;
-        noiseTable = new SimplexNoise(conditionSeed);
+
+    @In
+    ClimateConditionsSystem climateConditionsSystem;
+    @In
+    WorldProvider worldProvider;
+
+    public static final String TEMPERATURE = "temperature";
+    public static final String HUMIDITY = "humidity";
+
+    private String type;
+
+    public ConditionsBaseField(String type) {
+        this.type = type;
+        worldProvider = CoreRegistry.get(WorldProvider.class);
     }
 
-    public float get(float x, float y, float z) {
-        return TeraMath.clamp(getConditionAlpha(x, y, z), 0, 1);
+    public float get(float x, float y, float z, boolean clamp) {
+        if (clamp) {
+            return TeraMath.clamp(getConditionAlpha(x, y, z), 0f, 1f);
+        } else {
+            return getConditionAlpha(x, y, z);
+        }
     }
 
     private float getConditionAlpha(float x, float y, float z) {
-        float result = noiseTable.noise(x * noiseMultiplier, z * noiseMultiplier);
-        float temperatureBase = function.apply(TeraMath.clamp((result + 1.0f) / 2.0f));
-        if (y <= seaLevel) {
-            return temperatureBase;
-        } else if (y >= maxLevel) {
-            return 0;
+        if (type.equals(TEMPERATURE)) {
+            return ((float) worldProvider.getExtraData("coreWorlds.temperature", new Vector3i(x, y, z))) / 1000 - .001f * y;
```

Comment:
Is this extraData field actually registered somewhere? I can't find it if so. The main javadoc for ExtraBlockDataManager includes instructions on how to do this, and the BiomeManager in BiomesAPI is an example. Also, given that CoreWorlds doesn't actually use these extraData fields, and ClimateConditions does, it would make more sense to call them "climateConditions.temperature" and "climateConditions.humidity" instead.

Reply:
The extraData field still isn't actually registered.

Reply:
Isn't it registered by the last few lines of `SolidRasterizer`?

Reply:
By "`SolidRasterizer`", do you mean the class in CoreWorlds: https://github.com/Terasology/CoreWorlds/blob/develop/src/main/java/org/terasology/core/world/generator/rasterizers/SolidRasterizer.java ? I assume not, because the last few lines of that look not at all relevant, but there also isn't a `SolidRasterizer` defined in this module.

Reply:
Yes, in CoreWorlds with my open PR there [https://github.com/Terasology/CoreWorlds/pull/8](https://github.com/Terasology/CoreWorlds/pull/8) which adds the lines that I mention.

Reply:
I hadn't noticed that there was a CoreWorlds PR required for this. That would have been useful for you to mention in this PR's initial description. My previous comment that it would make more sense to call these extraData fields "climateConditions.(temperature/humidity)" rather than "coreWorlds.(temperature/humidity)" was made under the understanding that these fields would be introduced in this module. If they are actually introduced in CoreWorlds, then they should have "coreWorlds" not "climateConditions" in their names.